File: Loom.Core/Models/ConversationState.vb
Namespace Models
    ''' <summary>
    ''' Current conversation status
    ''' </summary>
    Public Class ConversationState
        Public Property TraceId As String = Guid.NewGuid().ToString()
        Public Property SystemPrompt As String
        Public Property TokenBudget As Integer? ' Null == unlimited
        Public Property Messages As New List(Of Message)
        Public Property TurnIndex As Integer = 0
    End Class
End Namespace

File: implementations/csharp/ConversationState.cs
using System.Collections.Generic;

namespace Models
{
    /// <summary>
    /// Current conversation status
    /// </summary>
    public class ConversationState
    {
        public string TraceId { get; set; }
        public string SystemPrompt { get; set; }
        public string TokenBudget { get; set; }
        public List<Message> Messages { get; set; }
        public int TurnIndex { get; set; } = 0;
    }
}
